Python/bytecodes.c

```diff
@@ -456,17 +456,15 @@ dummy_func(
             DISPATCH_INLINED(new_frame);
         }
 
-        // Alternative: (list, unused[oparg], v -- list, unused[oparg])
-        inst(LIST_APPEND, (v --)) {
-            PyObject *list = PEEK(oparg + 1);  // +1 to account for v staying on stack
-            ERROR_IF(_PyList_AppendTakeRef((PyListObject *)list, v) < 0, error);
+        // 'stuff' is a list object followed by (oparg - 1) unused values
+        inst(LIST_APPEND, (stuff[oparg], v -- stuff[oparg])) {
+            ERROR_IF(_PyList_AppendTakeRef((PyListObject *)stuff[0], v) < 0, error);
```

Comment:
This (and similar instructions that skip some stuff like `SET_ADD`, `SET_UPDATE`, `MAP_ADD`, `COPY`, `SWAP`, etc.) seem like they would be a bit cleaner as something like:

```suggestion
        inst(LIST_APPEND, (list, unused[oparg - 1], v -- list, unused[oparg - 1])) {            
            ERROR_IF(_PyList_AppendTakeRef((PyListObject *)list, v) < 0, error);
```

Then you don't need the comment, or to index into `stuff`. Otherwise, we're basically just doing `PEEK(...)` in a weird-ish upside down way.

(I see that you support multiplication already... probably for `BUILD_MAP`?)

Reply:
Yeah, I couldn't do without integer multiplication, but I didn't want to go implement full expression parsing (nor just skipping tokens until `]`). Maybe I'll reconsider (there were more opcodes like this than I expected).

Reply:
Maybe ~skipping~collecting tokens until `]` would be a reasonable approach? That could be solved (nearly) locally in parser.py, since the dimension is converted to a string anyway before it is passed to the `StackEffect` constructor.

Reply:
And done.